sway/desktop/transaction.c

```diff
@@ -345,9 +345,7 @@ static void set_instruction_ready(
 	}
 
 	instruction->container->instruction = NULL;
-	if (!txn_debug) {
-		transaction_progress_queue();
-	}
+	transaction_progress_queue();
```

Comment:
Note that `txn_debug` snuck back in after a rebase, and removing it completely is the correct behaviour. We now stall transactions (when using `-Dtxn-wait`) by inflating the `num_waiting` figure rather than avoiding calls to `transaction_progress_queue`.